vllm/engine/arg_utils.py

```diff
@@ -1210,6 +1213,60 @@ def create_engine_config(self) -> VllmConfig:
         )
 
 
+@dataclass
+class V1EngineArgs(EngineArgs):
+    """Arguments for vLLM engine v1."""
+
+    # V1's default values that differ from the default values in EngineArgs.
+    # This allows to switch between V1 and V0's default behaviour transparently.
+    enable_prefix_caching: bool = True
+    max_num_seqs: int = 1024
+    max_num_batched_tokens: Optional[int] = None
+
+    @staticmethod
+    def add_cli_args(parser: FlexibleArgumentParser) -> FlexibleArgumentParser:
+        parser = EngineArgs.add_cli_args(parser)
+        return parser
+
+    def create_engine_config(
+        self, usage_context: Optional[UsageContext] = None
+    ) -> VllmConfig:
+        assert (
+            usage_context is not None
+        ), "usage_context must be provided for V1EngineArgs"
```

Comment:
@WoosukKwon We need to pass usage_context because the default value depends on it, but this argument looks a bit weird to me. Do you have a better way to decide the default `max_num_batched_tokens`?